New compilation warning in release mode

In release mode with gcc 4.8 we get the following new warning on master

```
BUILD/DOC_ROOT/DDS/dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp: In member function ‘void OPENDDS_3_17_0::OpenDDS::DCPS::RtpsUdpDataLink::RtpsReader::gather_preassociation_acknack_i(OPENDDS_3_17_0::OpenDDS::DCPS::RtpsUdpDataLink::MetaSubmessageVec&, const WriterInfo_rch&)’:
BUILD/DOC_ROOT/DDS/dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp:1969:27: warning: unused variable ‘recvd’ [-Wunused-variable]
```

Which is coming from https://github.com/objectcomputing/OpenDDS/blob/37fae148bcc317aaef436f1e3c750e22ecfc3da7/dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp#L1970

--- a/dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp
+++ b/dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp
@@ -2016,8 +2016,7 @@ RtpsUdpDataLink::RtpsReader::gather_preassociation_acknack_i(MetaSubmessageVec& 
 {
   using namespace OpenDDS::RTPS;
 
-  const DisjointSequence& recvd = writer->recvd_;
-  OPENDDS_ASSERT(recvd.empty());
+  OPENDDS_ASSERT(writer->recvd_.empty());
   const CORBA::ULong num_bits = 0;
   const LongSeq8 bitmap;
   const EntityId_t reader_id = id_.entityId;

